frontend/src/lib/types.ts

```diff
@@ -0,0 +1,57 @@
+import subjectDetails from "../data/subjectDetails.json";
+import schedule from "../data/schedule.json";
+import emptySchedule from "../data/empty_schedule.json";
+
+export type Schedule = typeof schedule;
+
+export type SubjectDetails = typeof subjectDetails;
+
+export type EmptySchedule = typeof emptySchedule;
+
+export type Complex = "Any" | "NC" | "NR" | "Vikramshila";
+export type Floor = "Any" | "1" | "2" | "3" | "4";
+
+export type ChillPlace = {
```

Comment:
Can use a better type name like `Classroom`

P.S., I don't see this type used anywhere. Let's merge `isWeekend` attribute here itself and rename to `Classroom`